iceoryx_examples/waitset/README.md

```diff
@@ -1,6 +1,586 @@
+{
+    subscriberVector[i].attachTo(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES, FIRST_GROUP_ID);
+}
+
+for (auto i = 2; i < 4; ++i)
+{
+    subscriberVector[i].attachTo(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES, SECOND_GROUP_ID);
+}
+```
+
+The event loop calls `auto triggerVector = waitset.wait()` in a blocking call to
+receive a vector of all the _Triggers_ which were triggered. If the _Trigger_
+did originate from the `shutdownTrigger` we terminate the program.
+```cpp
+while (true)
+{
+    auto triggerVector = waitset.wait();
+
+    for (auto& trigger : triggerVector)
+    {
+        if (trigger.doesOriginateFrom(&shutdownTrigger))
+        {
+            return (EXIT_SUCCESS);
+        }
+```
+
+The remaining part of the loop is handling the subscribers. For the first group
+we would like to print the received data to the console and for the second group
+we just dismiss the received data.
+```cpp
+else if (trigger.getTriggerId() == FIRST_GROUP_ID)
+{
+    auto subscriber = trigger.getOrigin<iox::popo::UntypedSubscriber>();
+    subscriber->take().and_then([&](iox::popo::Sample<const void>& sample) {
+        const CounterTopic* data = reinterpret_cast<const CounterTopic*>(sample.get());
+        std::cout << "received: " << std::dec << data->counter << std::endl;
+    });
+}
+else if (trigger.getTriggerId() == SECOND_GROUP_ID)
+{
+    std::cout << "dismiss data\n";
+    auto subscriber = trigger.getOrigin<iox::popo::UntypedSubscriber>();
+    subscriber->releaseQueuedSamples();
+}
+```
+**Important** The second group needs to release all queued samples otherwise
+the WaitSet would notify the user again that the subscriber from the second 
+group has new samples.
+
+### Individual
+When every _Triggerable_ requires a different reaction we need to know the 
+origin of a _Trigger_. We can call `trigger.doesOriginateFrom(TriggerOrigin)`
+which will return true if the trigger originates from _TriggerOrigin_ and
+otherwise false.
+
+We start this example by creating a _WaitSet_ and attaching the `shutdownTrigger`
+to handle `CTRL-c`.
+```cpp
+iox::popo::WaitSet waitset;
+
+shutdownTrigger.attachTo(waitset);
+```
+
+Additionally, we create two subscribers, subscribe them to our topic and attach
+them to the waitset to let them inform us whenever they receive a new sample.
+```cpp
+iox::popo::TypedSubscriber<CounterTopic> subscriber1({"Radar", "FrontLeft", "Counter"});
+iox::popo::TypedSubscriber<CounterTopic> subscriber2({"Radar", "FrontLeft", "Counter"});
+
+subscriber1.subscribe();
+subscriber2.subscribe();
+
+subscriber1.attachTo(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES);
+subscriber2.attachTo(waitset, iox::popo::SubscriberEvent::HAS_NEW_SAMPLES);
+```
+
+With that set up we enter the event loop and handle the program termination
+first.
+```cpp
+while (true)
+{
+    auto triggerVector = waitset.wait();
+
+    for (auto& trigger : triggerVector)
+    {
+        if (trigger.doesOriginateFrom(&shutdownTrigger))
```

Comment:
At the moment I do not require a stateful function wrapper for this particular case. 
For the readme I would like to have a very verbose examples so that you can go and read a particular use case without reading the preceding ones.